### PR TITLE
frontend: Add the ability to specify the cluster in apiProxy methods

### LINKED ISSUE #1472: Allow to pass the cluster as an argument to resource management functions
When we list/get resources, the cluster is taken from the context we are in.
However, given we have the new Home view, where we are not in a cluster context and yet maybe plugin devs want to show data for each cluster, we should have a way to pass the cluster as an argument for resource related functions.

A version of this is implemented in the aggregated-view branch already, so we should clean it up and publish it.

fixes #1459

### PR DESCRIPTION
**Pull request by joaquimrocha**
We need ot be able to specify the cluster so we can later call the
apiProxy related methods on any cluster instead of just the currently
set up one.

fixes #1472

### CODE REVIEW

File: frontend/src/lib/k8s/apiProxy.ts [hunk 1]
```diff
@@ -607,23 +616,30 @@ function singleApiFactory(group: string, version: string, resource: string) {
...
+    ) => {
       if (isDebugVerbose('k8s/apiProxy@singleApiFactory list')) {
-        console.debug('k8s/apiProxy@singleApiFactory list', { queryParams });
+        console.debug('k8s/apiProxy@singleApiFactory list', { cluster, queryParams });
       }
 
-      return streamResults(url, cb, errCb, queryParams);
+      return streamResultsForCluster(url, { cb, errCb, cluster }, queryParams);
     },
     get: (name: string, cb: StreamResultsCb, errCb: StreamErrCb, queryParams?: QueryParameters) =>
```
Comment (illume) [inline frontend/src/lib/k8s/apiProxy.ts:631]: I wonder why the GET don't get a cluster?

**Review by illume — COMMENTED** [review]
I'm not sure how to test this really. Just reading the code it looks ok.

We really should have some sort of tests for this code in the future... maybe an example plugin using the cluster name parts. Or using the functionality somehow in the app.

File: frontend/src/lib/k8s/apiProxy.ts (hunk 1 above)
Comment (joaquimrocha) [reply frontend/src/lib/k8s/apiProxy.ts:631]: The main motivation was the listing functions. I think it makes sense to have the `get` and other functions receive the cluster too.
Let's do that in a later iteration.

**Comment by joaquimrocha** [comment]
I have added the cluster arg to the missing functions but noticed that getting CRs is broken. Making this WIP for further fixes.

**Comment by joaquimrocha** [comment]
I have found and fixed the issue I had encountered before. Should be good to review.
I really wish this is the last time we have to touch apiProxy and that we very soon implement a saner version of it.

File: frontend/src/lib/k8s/apiProxy.ts [hunk 2]
```diff
@@ -852,38 +907,52 @@ export function post(
...
-  return request(url, opts, autoLogoutOnAuthError);
+  const cluster = getCluster() || '';
+  const opts = { method: 'POST', body, headers: JSON_HEADERS, cluster, ...requestOptions };
+
+  return clusterRequest(url, { ...opts, autoLogoutOnAuthError });
 }
 
 export function patch(url: string, json: any, autoLogoutOnAuthError = true, requestOptions = {}) {
   const body = JSON.stringify(json);
+  const cluster = getCluster() || '';
```
Comment (illume) [inline frontend/src/lib/k8s/apiProxy.ts:921]: This one is not using the clusterName from requestOptions.

File: frontend/src/lib/k8s/apiProxy.ts [hunk 3]
```diff
@@ -852,38 +907,52 @@ export function post(
...
+    cluster: clusterName || getCluster() || '',
+    ...restOptions,
+  };
+  return clusterRequest(url, opts);
 }
 
 export function remove(url: string, requestOptions = {}) {
-  const opts = { method: 'DELETE', headers: JSON_HEADERS, ...requestOptions };
-  return request(url, opts);
+  const cluster = getCluster() || '';
```
Comment (illume) [inline frontend/src/lib/k8s/apiProxy.ts:953]: This one is not using the clusterName from requestOptions.

File: frontend/src/lib/k8s/apiProxy.ts (hunk 1 above)
Comment (illume) [reply frontend/src/lib/k8s/apiProxy.ts]: ok

**Comment by illume** [comment]
> We really should have some sort of tests for this code in the future... maybe an example plugin using the cluster name parts. Or using the functionality somehow in the app.

@joaquimrocha mentioned he is working on replacing this code in a separate piece of work. We agreed that tests for this can be skipped and we put extra time into manually testing this.

**Pull request merged**